### ISSUE TITLE
Bad argument #1 to 'copy' (sizes do not match) at model:forward(input:cuda()) 

### ISSUE THREAD

**Issue by ttoinou** [DmitryUlyanov/texture_nets#24]
Hi I have this error at line 33 of test.lua : 
```
torch/install/share/lua/5.1/nn/Concat.lua:25: bad argument #1 to 'copy' (sizes do not match at /tmp/luarocks_cutorch-scm-1-9130/cutorch/lib/THC/THCTensorCopy.cu:30)
stack traceback:
	[C]: in function 'copy'

```
at the line 33 of test.lua when doing `model:forward(input:cuda()) `
It's doing that with my own model I have pretrained with train.lua, it's not doing that with the pretrained model downloaded from the readme...

How do I use my own pretrained styles ?

Thanks in advance :) .

**Comment by paperkettle** [comment]
+1 having the same problem! 
test.lua works for me when I use the supplied model but then not with my own models. 
It appears I'm successfully training? 
train.lua outputs checkpoint models (and I watch style transfer results in the web display while they're being made). 
But when I try to run test.lua using those models I see this error: 

`/root/torch/install/bin/luajit: /root/torch/install/share/lua/5.1/nn/Container.lua:67: 
In 1 module of nn.Sequential:
In 1 module of nn.Concat:
In 1 module of nn.Sequential:
In 1 module of nn.Concat:
In 1 module of nn.Sequential:
In 1 module of nn.Concat:
In 1 module of nn.Sequential:
In 1 module of nn.Concat:
In 1 module of nn.Sequential:
/root/torch/install/share/lua/5.1/nn/Concat.lua:25: bad argument #1 to 'copy' (sizes do not match at /tmp/luarocks_cutorch-scm-1-3772/cutorch/lib/THC/THCTensorCopy.cu:31)
stack traceback:
	[C]: in function 'copy'
	/root/torch/install/share/lua/5.1/nn/Concat.lua:25: in function </root/torch/install/share/lua/5.1/nn/Concat.lua:9>
	[C]: in function 'xpcall'
	/root/torch/install/share/lua/5.1/nn/Container.lua:63: in function 'rethrowErrors'
	/root/torch/install/share/lua/5.1/nn/Sequential.lua:44: in function </root/torch/install/share/lua/5.1/nn/Sequential.lua:41>
	[C]: in function 'xpcall'
	/root/torch/install/share/lua/5.1/nn/Container.lua:63: in function 'rethrowErrors'
	/root/torch/install/share/lua/5.1/nn/Concat.lua:12: in function </root/torch/install/share/lua/5.1/nn/Concat.lua:9>
	[C]: in function 'xpcall'
	/root/torch/install/share/lua/5.1/nn/Container.lua:63: in function 'rethrowErrors'
	...
	[C]: in function 'xpcall'
	/root/torch/install/share/lua/5.1/nn/Container.lua:63: in function 'rethrowErrors'
	/root/torch/install/share/lua/5.1/nn/Concat.lua:12: in function </root/torch/install/share/lua/5.1/nn/Concat.lua:9>
	[C]: in function 'xpcall'
	/root/torch/install/share/lua/5.1/nn/Container.lua:63: in function 'rethrowErrors'
	/root/torch/install/share/lua/5.1/nn/Sequential.lua:44: in function 'forward'
	test.lua:33: in main chunk
	[C]: in function 'dofile'
	/root/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
	[C]: at 0x00406670

WARNING: If you see a stack trace below, it doesn't point to the place where this error occured. Please use only the one above.
stack traceback:
	[C]: in function 'error'
	/root/torch/install/share/lua/5.1/nn/Container.lua:67: in function 'rethrowErrors'
	/root/torch/install/share/lua/5.1/nn/Sequential.lua:44: in function 'forward'
	test.lua:33: in main chunk
	[C]: in function 'dofile'
	/root/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
	[C]: at 0x00406670
`

**Comment by paperkettle** [comment]
got it working by using the johnson model rather than pyramid. seems like maybe the pyramid doesn't match up with the test.lua code? will look into this more but any extra info around this would be very helpful!

**Comment by xpeng** [comment]
same problem, but i use smaller image size like "-image_size 256", it's work.
is it possible error on gpu memory? like something "out of memory"

**Comment by paperkettle** [comment]
I don't think it is memory size problem - in my limited experience so far, those tend to output other kinds of errors. The machine I'm on has 8gb of video ram and can do 1600x1600 images no problem using the Johnson model (and Johnson's style transfer code as well). Seems to have something to do with the (default) pyramid model for me. I should try the other one that the code ships with as well. I don't have an intuition yet for what kinds of different results one would from johnson vs pyramid vs skip_unpool, but I'm curious if others can chime in!

**Comment by htoyryla** [comment]
Same problem here. To me it appears related to the shape of the image. If I resize the input image to a square, e.g. 800x800 it works.

**Comment by DmitryUlyanov** [comment]
Yes, due to pooling layers the width and height should be a multiple of 32.

**Comment by alexblickenstaff** [comment]
You could try resizing both dimensions to the nearest multiple of 32. For example, if you have an 800x600 image, instead of resizing to 800x800 you could resize to 800x608 and it should work.

**Comment by htoyryla** [comment]
"You could try resizing both dimensions to the nearest multiple of 32. For example, if you have an 800x600 image, instead of resizing to 800x800 you could resize to 800x608 and it should work."

Yes, from Dmitry's answer I assumed this would work, but I haven't tried it yet; to make it really useful in practice I should implement the scaling in code but have other things going on now. So far I have used the Johnson model instead.

**Comment by alexblickenstaff** [comment]
You can replace the image scaling code [here](https://github.com/DmitryUlyanov/texture_nets/blob/master/test.lua#L36-L39) with
```
local img = image.load(params.input_image, 3)
local w = img:size(2)
local h = img:size(3)
local newW = math.floor(w/32 + 0.5) * 32
local newH = math.floor(h/32 + 0.5) * 32
img = image.scale(img, newW, newH)
```

**Comment by htoyryla** [comment]
Yes, of course.
